# Notebook Instructions

1. If you are new to Jupyter notebooks, please go through this introductory manual <a href='https://quantra.quantinsti.com/quantra-notebook' target="_blank">here</a>.
1. Any changes made in this notebook would be lost after you close the browser window. **You can download the notebook to save your work on your PC.**
1. Before running this notebook on your local PC:<br>
i.  You need to set up a Python environment and the relevant packages on your local PC. To do so, go through the section on "**Run Codes Locally on Your Machine**" in the course.<br>
ii. You need to **download the zip file available in the last unit** of this course. The zip file contains the data files and/or python modules that might be required to run this notebook.

## Expressions

'Expressions' are generally a combination of numbers, variables, and operators. <br>
<br>
In this IPython notebook, we will make use of Expressions to understand the TVM concepts. 

### Future value (FV)

What would be the FV, if I have $1000 with me now and I will be investing it for 1 year, at an annual return of 5%? 

In [2]:
PV = 1000
r = 0.05
n = 1

FV = PV * ((1+r) ** n) # Formula for calculating Future Value

print (FV)

1050.0


### Present value (PV)

What would be the PV, if I have to discount $1050 at a 5% annual rate for a period of 1 year?

In [3]:
FV = 1050
r = 0.05
n = 1

PV = FV / ((1 + r) ** n) # Formula for calculating Present Value

print (PV)

1000.0


### Compounding 

Assume that the 5% annual interest rate bond makes semiannual payments. That is, for an investment of $1000, you will get 25 dollars, after the first 6 months and another 25 dollars after 1 year. The annual rate of interest is 5%. What would be the FV, if I hold the bond for 1 year? 

In [4]:
PV = 1000
r = 0.05
n = 2 # number of periods = 2 since bond makes semiannual payments
t = 1 # number of years

FV = PV * ((1+(r/n)) ** (n*t)) # Formula for compounding

print (FV)

1050.625


### Annuity payments

What would be the annual periodic saving amount, if you want a lump sum of $9476.96 at the end of 3 years? The rate of return is 10%? (Assuming the first payment is made at the start of each year).<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [1]:
r = 0.1
n = 3
PV = 0
FV = 9476.96

AP = (FV * r) / (((1 + r) ** n - 1)*(1+r)) # Formula for Annuity payments, given Future Value

print (AP)

2602.8453721505043


What would be the PV, given a cash outflow of $2500 for a period of 5 years and rate of return being 10%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [6]:
r = 0.1
n = 5
AP = 2500

PV = (AP * (1 - ((1 + r) ** -n))) / r # Formula for PV, given Annuity payments

print (PV)

9476.966923521126


What would be the PV, given a cash outflow of $30,000 for a period of 45 years and rate of return being 8%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [7]:
r = 0.08
n = 45
AP1 = 30000

PV = (AP1 * (1 - ((1 + r) ** -n))) / r # Formula for PV, given Annuity payments

print (PV)

363252.0450945144


What would be the annual saving amount (AP), if you want to save a lump sum of $363252.045 in 25 years and rate of return being 15%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [8]:
r = 0.15
n = 25
PV = 0
FV = 363252.045095

AP = (FV * r) / (((1 + r) ** n - 1)*(1+r)) # Formula to calculate Annuity Payments, given FV

#AP = (r * PV) / (1 - ((1 + r) ** -n)) # Formula to calculate Annuity Payments, given PV

print (AP)

1484.4065248166885


These are some ways, one can use Expressions. <br>

In the upcoming section, we will learn about Python <b>Data Structures</b>.<br><br>